pghoard/basebackup.py

```diff
@@ -710,7 +710,11 @@ def create_and_upload_chunks(
                     # and this assumption greatly simplifies the logic.
                     task_to_wait = pending_compress_and_encrypt_tasks.pop(0)
                     chunk_files.append(task_to_wait.result())
-
+                    self.metrics.gauge(
+                        "pghoard.basebackup_estimated_progress",
+                        len(chunk_files) * 100 / len(chunks),
```

Comment:
Sorry one more thing: Looks like the statsd implementation in pghoard doesn't explicitly define the type for this and some of the values may end up being integers and some floats, which could be problematic in some cases. Explicit `float(len(chunk_files) * 100 / len(chunks))` would be good to ensure metric type is consistent. Same for the other call.